lib/addressable/uri.rb

```diff
@@ -53,7 +53,7 @@ module CharacterClasses
       PCHAR = (UNRESERVED + SUB_DELIMS + "\\:\\@").freeze
       SCHEME = (ALPHA + DIGIT + "\\-\\+\\.").freeze
       HOST = (UNRESERVED + SUB_DELIMS + "\\[\\:\\]").freeze
-      AUTHORITY = (PCHAR + "\\[\\:\\]").freeze
+      AUTHORITY = (PCHAR + "\\[\\]").freeze
```

Comment:
Unrelated minor fix to remove the following warning: `uri.rb:417: warning: character class has duplicated range: /[^a-zA-Z0-9\-\.\_\~\!\$\&\'\(\)\*\+\,\;\=\:\@\[\:\]]` because the column `:` is already in `PCHAR`

Reply:
Isn't this already done in https://github.com/sporkmonger/addressable/pull/438?

Reply:
Ah indeed, I must have forked from and outdated local copy, thanks for pointing this out I'll rebase :)

Reply:
Actually I was up to date, the column is back and I'm not even sure why :thinking:. I checked the git commit log, I can see this commit which remove the column: dd84176c2ad6c71bbec4b66bfc694d80969149ae and then this commit: 178bcd97fc77b342a3bda6f5b6e17f8ea344bd51 added a `freeze` but somehow the column is back, even in the "previous" version in red...

`git log -p lib/addressable/uri.rb`:
![image](https://user-images.githubusercontent.com/201687/218703497-c9a501b2-dba1-48f5-a2ac-f847c5c85f0b.png)

I dont... :shrug: Any idea how this could happen? Anyway I think we can keep this change to put it back in the codebase then.

Reply:
I #449 was opened the same day #438 was merged so #449 probably didn't have the latest changes and we did not notice in review 😞

Reply:
Plus GitHub not understanding it was a conflict? Anyway, I asked @sporkmonger to double check repo settings in that area (as I don't have access to that myself)

Reply:
Weirdest to me is that it's not only Github here, even in plain git (as in my git log screenshot) we can clearly see the "before" (red) line in https://github.com/sporkmonger/addressable/commit/178bcd97fc77b342a3bda6f5b6e17f8ea344bd51 being different than the "after" (green) line in https://github.com/sporkmonger/addressable/commit/dd84176c2ad6c71bbec4b66bfc694d80969149ae. And there's no other comits on this file in between. I have no idea how this can even be possible.

Anyway I suppose it's ok to keep this change here then.

Reply:
Yeah we'll keep it here 

The merge commit diff is correct: https://github.com/sporkmonger/addressable/commit/7ecf75146c1a53b089150ccc33c909214d728323

Reply:
Interesting, I don't see this commit in the git log for `lib/addressable/uri.rb`: https://github.com/sporkmonger/addressable/commits/main/lib/addressable/uri.rb

And when I check it with:
```bash
> git show 7ecf75146c1a53b089150ccc33c909214d728323
commit 7ecf75146c1a53b089150ccc33c909214d728323
Merge: 068f673 41f12dd
Author: Bob Aman <sporkmonger@users.noreply.github.com>
Date:   Fri Jul 29 20:35:38 2022 -0700

    Merge pull request #449 from okeeblow/freeze_concatenated_strings
    
    Explicitly freeze concatenated-`String` constants to unbreak on non-main `Ractor`s.

>
```
It's "empty", there's no file changed. There's probably many things I still don't get about `git` :smiling_face_with_tear:

Reply:
Yeah, I just [googled this](https://stackoverflow.com/questions/43138569/how-to-get-git-log-p-to-show-changes-in-merge-commits) and I learned something too here :) with `-m` (or the longer variant `--diff-merges=on`) `git log`/`git show` shows me the diff

```
git log -1 --diff-merges=on 7ecf751
git show --diff-merges=on 7ecf751
```

Reply:
Oh interesting, thanks :)